src/main/java/software/bernie/geckolib3/renderers/geo/ExtendedGeoEntityRenderer.java

```diff
@@ -0,0 +1,375 @@
+				e1.printStackTrace();
+			}
+		} catch (SecurityException e) {
+			e.printStackTrace();
+		}
+		if(FIELD_CUBES != null) {
+			FIELD_CUBES.setAccessible(true);
+		}
+	}
+	
+	protected float widthScale;
+	protected float heightScale;
+
+	public final Function<T, ResourceLocation> TEXTURE_GETTER;
+	public final Function<T, ResourceLocation> MODEL_ID_GETTER;
+
+	protected ExtendedGeoEntityRenderer(EntityRendererManager renderManager, AnimatedGeoModel<T> modelProvider) {
+		this(renderManager, modelProvider, 1F, 1F, 0);
+	}
+
+	@SuppressWarnings("resource")
+	protected void bindTexture(ResourceLocation textureLocation) {
+		Minecraft.getInstance().textureManager.bind(textureLocation);
+	}
+
+	protected ExtendedGeoEntityRenderer(EntityRendererManager renderManager, AnimatedGeoModel<T> modelProvider, float widthScale, float heightScale, float shadowSize) {
+		super(renderManager, modelProvider);
+
+		this.MODEL_ID_GETTER = modelProvider::getModelLocation;
+		this.TEXTURE_GETTER = modelProvider::getTextureLocation;
+
+		this.shadowRadius = shadowSize;
+		this.widthScale = widthScale;
+		this.heightScale = heightScale;
+	}
+
+	/*
+	 * 0 => Normal model 1 => Magical armor overlay
+	 */
+	private int currentModelRenderCycle = 0;
+
+	// Entrypoint for rendering, calls everything else
+	@Override
+	public void render(T entity, float entityYaw, float partialTicks, MatrixStack stack, IRenderTypeBuffer bufferIn, int packedLightIn) {
+		this.currentModelRenderCycle = 0;
+		super.render(entity, entityYaw, partialTicks, stack, bufferIn, packedLightIn);
+	}
+
+	// Rendercall to render the model itself
+	@Override
+	public void render(GeoModel model, T animatable, float partialTicks, RenderType type, MatrixStack matrixStackIn, IRenderTypeBuffer renderTypeBuffer, IVertexBuilder vertexBuilder, int packedLightIn, int packedOverlayIn, float red, float green,
+			float blue, float alpha) {
+		super.render(model, animatable, partialTicks, type, matrixStackIn, renderTypeBuffer, vertexBuilder, packedLightIn, packedOverlayIn, red, green, blue, alpha);
+		this.currentModelRenderCycle++;
+	}
+
+	protected float getWidthScale(T entity) {
+		return this.widthScale;
+	}
+
+	protected float getHeightScale(T entity) {
+		return this.heightScale;
+	}
+
+	@Override
+	public void renderEarly(T animatable, MatrixStack stackIn, float ticks, IRenderTypeBuffer renderTypeBuffer, IVertexBuilder vertexBuilder, int packedLightIn, int packedOverlayIn, float red, float green, float blue, float partialTicks) {
+		this.rtb = renderTypeBuffer;
+		super.renderEarly(animatable, stackIn, ticks, renderTypeBuffer, vertexBuilder, packedLightIn, packedOverlayIn, red, green, blue, partialTicks);
+		if (this.currentModelRenderCycle == 0 /* Pre-Layers */) {
+			float width = this.getWidthScale(animatable);
+			float height = this.getHeightScale(animatable);
+			stackIn.scale(width, height, width);
+		}
+	}
+
+	@Override
+	public ResourceLocation getTextureLocation(T entity) {
```

Comment:
This is unnecessary, see comment above about storing the model instead of references to the functions.